source/slang/slang-check-decl.cpp

```diff
@@ -7833,6 +8045,73 @@ namespace Slang
         return as<SeqStmt>(stmt->body);
     }
 
+    static bool _doesCtorExpectInitializerListUsage(ConstructorDecl* ctor)
```

Comment:
This excludes user defined ctors, accorading to the name of this function, it means user defined ctors does not expect initializer list usage? That seems wrong. What exactly are we testing for here?

Reply:
This function should be renamed to "_isCtorASynthisizedMemberwiseCtor".

The goal is to detect which functions are member-wise ctor's and need their body filled.

Reply:
We already have a `ToBeSynthesized` modifier that we can use for this purpose.

Reply:
Addressed in this change:
https://github.com/shader-slang/slang/pull/5107/commits/5771007f0f14dad1d69c1d499758d10fa24bc087

I found out that this function is not only check whether a function is synthesized, but also this is a member-wise ctor, so only checking `ToBeSynthesized` could not be enough.